tensorflow/contrib/cmake/patches/protobuf/build_pb.py

```diff
@@ -0,0 +1,64 @@
+import sys
+import os
+import subprocess
+from pathlib import Path
+
+
+def generate_proto(source, require = True, protoc = None):
```

Comment:
Can you add descriptions for arguments, for e.g. I can't tell what "require" does just from reading the signature?
Even better if you can write add the comment in the following format:
"""One line description.

Args:
  arg name: description for arg
"""

Also, can you change protoc to an argument without default value? Seems like protoc is required. Having a default value makes it seems like it is ok to call 'generate_proto' without 'protoc'.

Reply:
Ok, but actually this is copied from `setup.py` file in protobuf, and I wonder why this failure is ignored for so long? 
Maybe It's better to require `protobuf` to be installed before building the tensorflow python packet ?